app/src/main/java/org/stepik/android/adaptive/pdd/ui/fragment/RatingFragment.kt

```diff
@@ -0,0 +1,86 @@
+package org.stepik.android.adaptive.pdd.ui.fragment
+
+import android.os.Bundle
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.DividerItemDecoration
+import android.support.v7.widget.LinearLayoutManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import org.stepik.android.adaptive.pdd.R
+import org.stepik.android.adaptive.pdd.core.presenter.BasePresenterFragment
+import org.stepik.android.adaptive.pdd.core.presenter.PresenterFactory
+import org.stepik.android.adaptive.pdd.core.presenter.RatingPresenter
+import org.stepik.android.adaptive.pdd.core.presenter.contracts.RatingView
+import org.stepik.android.adaptive.pdd.databinding.FragmentRatingBinding
+import org.stepik.android.adaptive.pdd.ui.adapter.RatingAdapter
+
+class RatingFragment : BasePresenterFragment<RatingPresenter, RatingView>(), RatingView {
+    private lateinit var binding: FragmentRatingBinding
+
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = FragmentRatingBinding.inflate(inflater, container, false)
+        binding.recycler.layoutManager = LinearLayoutManager(context)
+
+        val divider = DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
+        divider.setDrawable(ContextCompat.getDrawable(context, R.drawable.stroke))
+        binding.recycler.addItemDecoration(divider)
+
+        val spinnerAdapter = ArrayAdapter<CharSequence>(context, R.layout.item_rating_period, context.resources.getStringArray(R.array.rating_periods))
+        spinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        binding.spinner.adapter = spinnerAdapter
+
+        binding.spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(p0: AdapterView<*>?) {}
+
+            override fun onItemSelected(p0: AdapterView<*>?, p1: View?, pos: Int, p3: Long) {
+                presenter?.changeRatingPeriod(pos)
+            }
+        }
+
+        binding.tryAgain.setOnClickListener { presenter?.retry() }
+        return binding.root
+    }
+
+    override fun onLoading() {
+        binding.error.visibility = View.GONE
+        binding.progress.visibility = View.VISIBLE
+        binding.container.visibility = View.GONE
+    }
+
+    override fun onError() {
+        binding.error.visibility = View.VISIBLE
```

Comment:
Here if the server will be shutted down, the user will see "No internet". Let's handle it and do not show strange messages to user. We need to classify error: no internet/request failed. In the 1st case show error, in the second show nothing and just log this to Analytic.